zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -323,6 +326,29 @@ protected void sendAndReceive(
         parent.performScannerHookAfterScan(message, this);
     }
 
+    /**
+     * Updates the Content-Length header of the request.
+     *
+     * <p>For methods with no nor expected enclosed content the header is removed otherwise in all
+     * other cases the header is updated to match the length of the content.
+     *
+     * @param message the message to update.
+     * @since 2.12.0
+     */
+    protected void updateRequestContentLength(HttpMessage message) {
+        int bodyLength = message.getRequestBody().length();
+        String method = message.getRequestHeader().getMethod();
+        if (bodyLength == 0
```

Comment:
If it has a body but is one of these methods it’ll still be updated. Should the unit tests include that?

Reply:
That's `shouldAddContentLengthHeaderWhenNeeded`.

Reply:
`shouldAddContentLengthHeaderWhenNeeded `is the case I was thinking. Thanks! I shoulda skipped around instead of just top downing it :grinning: